smack-extensions/src/main/java/org/jivesoftware/smackx/geoloc/GeoLocationManager.java

```diff
@@ -83,4 +89,73 @@ public static boolean isGeoLocationMessage(Message message) {
         return GeoLocation.from(message) != null;
     }
 
+    /**
+     * Send geolocation through the PubSub node.
+     * 
+     * @param geoLocation
+     * @throws InterruptedException
+     * @throws NotConnectedException
+     * @throws XMPPErrorException
+     * @throws NoResponseException
+     */
+    public void sendGeolocation(GeoLocation geoLocation)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        getNode().send(new PayloadItem<GeoLocation>(geoLocation));
+    }
+
+    /**
+     * Send empty geolocation through the PubSub node.
+     * 
+     * @throws InterruptedException
+     * @throws NotConnectedException
+     * @throws XMPPErrorException
+     * @throws NoResponseException
+     */
+    public void stopPublishingGeolocation()
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        GeoLocation emptyGeolocation = new GeoLocation.Builder().build();
+        getNode().send(new PayloadItem<GeoLocation>(emptyGeolocation));
+    }
+
+    /**
+     * Returns the geolocation node in PubSub.
+     * 
+     * @return the geolocation PubSub node
+     * @throws InterruptedException
+     * @throws NotConnectedException
+     * @throws XMPPErrorException
+     * @throws NoResponseException
+     */
+    public LeafNode getNode()
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        PubSubManager pubSubManager = PubSubManager.getInstance(connection());
+        LeafNode node;
+
+        try {
+            node = pubSubManager.getNode(GeoLocation.NAMESPACE);
```

Comment:
It appears that comment got lost: Please create a `PubSubManager.getOrCreateNode(String)` method, as it has become appearant that such functionality is required by multiple Managers (and therefore also something PubSub users would be interested in)